### ISSUE TITLE
last node-v8 REPL issue

### ISSUE THREAD

**Issue by vsemozhetbyt** [nodejs/node-v8#138]
* **Version**: v14.0.0-v8-canary2020011995087de57b
* **Platform**: Windows 7 x64
* **Subsystem**: REPL

This is OK in 13.6.0 and v14.0.0-nightly20200119b318926634, but throws in v14.0.0-v8-canary2020011995087de57b:

```
> const foo = 1;
Uncaught SyntaxError: Identifier 'foo' has already been declared
```
It seems all the intermediate evaluations are saved in the context. See:
```
> let abc
Uncaught SyntaxError: Identifier 'abc' has already been declared
> z
Uncaught ReferenceError: z is not defined
> a
undefined
> ab
undefined
> abc
undefined
>
```

**Comment by cclauss** [comment]
```
> foo
???
```

**Comment by vsemozhetbyt** [comment]
```
> const foo = 1;
Uncaught SyntaxError: Identifier 'foo' has already been declared
> foo
1
>
```

**Comment by cclauss** [comment]
BEFORE you attempt to (re)define it, `foo` is already set to `1`?

**Comment by vsemozhetbyt** [comment]
No:
```
Welcome to Node.js v14.0.0-v8-canary2020011995087de57b.
Type ".help" for more information.
> foo
Uncaught ReferenceError: foo is not defined
>
```

**Comment by cclauss** [comment]
OK.  So there is no bug here.  You can defined a const once and only once in a run.  If you want to redefine a const then you need to reboot.

**Comment by vsemozhetbyt** [comment]
But i do not redefine the const:
```
Welcome to Node.js v14.0.0-v8-canary2020011995087de57b.
Type ".help" for more information.
> foo
Uncaught ReferenceError: foo is not defined
> const foo = 1
Uncaught SyntaxError: Identifier 'foo' has already been declared
```

**Comment by cclauss** [comment]
Please try same with `bar`.

**Comment by vsemozhetbyt** [comment]
The same in  13.6.0 and v14.0.0-nightly20200119b318926634:
```
Welcome to Node.js v13.6.0.
Type ".help" for more information.
> foo
Uncaught ReferenceError: foo is not defined
> const foo = 1
undefined
>
```

```
Welcome to Node.js v14.0.0-nightly20200119b318926634.
Type ".help" for more information.
> foo
Uncaught ReferenceError: foo is not defined
> const foo = 1
undefined
>
```

**Comment by vsemozhetbyt** [comment]
```
Welcome to Node.js v14.0.0-v8-canary2020011995087de57b.
Type ".help" for more information.
> bar
Uncaught ReferenceError: bar is not defined
> const bar = 1
Uncaught SyntaxError: Identifier 'bar' has already been declared
>
```

**Comment by vsemozhetbyt** [comment]
Also notable:
Google Chrome 81.0.4029.3 Dev with V8 8.1.187:
```
foo
VM46:1 Uncaught ReferenceError: foo is not defined
    at <anonymous>:1:1
(anonymous) @ VM46:1
const foo = 1
undefined 
```

Google Chrome 81.0.4031.0 canary with V8 8.1.218:
```
foo
VM36:1 Uncaught ReferenceError: foo is not defined
    at <anonymous>:1:1
(anonymous) @ VM36:1
const foo = 1
VM68:1 Uncaught SyntaxError: Identifier 'foo' has already been declared 
```
It seems something happens between V8 8.1.187 and V8 8.1.218 (the last node-v8 canary has 8.1.222).

**Comment by vsemozhetbyt** [comment]
Is this the same issue? https://bugs.chromium.org/p/chromium/issues/detail?id=1043151

**Comment by vsemozhetbyt** [comment]
In Chrome, this seems fixed at least since V8 8.1.281.

**Comment by vsemozhetbyt** [comment]
Seems fixed in [14.0.0-v8-canary202002123546737a75](https://nodejs.org/download/v8-canary/v14.0.0-v8-canary202002123546737a75/). Thank you.
